### PR TITLE
Add remaining 0.6 news items

### LINKED ISSUE #21475: Make sure NEWS.md lists all 0.6 deprecations and breaking changes
Reviewing the checklist I had in https://github.com/JuliaLang/julia/issues/17418 for the 0.5 release, pretty much the only thing we're missing for an 0.6 RC ~~(other than #21470)~~ is doing a final pass over NEWS.md and making sure all the deprecations and breaking changes that should be listed there are.

Check https://github.com/JuliaLang/julia/issues?utf8=%E2%9C%93&q=label%3Anews and https://github.com/JuliaLang/julia/issues?utf8=%E2%9C%93&q=label%3Adeprecation and https://github.com/JuliaLang/julia/blob/release-0.6/base/deprecated.jl#L117-L1325, make sure everything is listed.

### PR DESCRIPTION
**Pull request by ararslan**
This PR adds NEWS items for the remaining points in https://github.com/JuliaLang/julia/issues/21475#issuecomment-302899987, so I believe this fixes #21475.

Edit: Forgot to skip CI. Sorry about that.

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -831,14 +853,27 @@ Command-line option changes
...
 [#20543]: https://github.com/JuliaLang/julia/issues/20543
 [#20575]: https://github.com/JuliaLang/julia/issues/20575
 [#20609]: https://github.com/JuliaLang/julia/issues/20609
 [#20889]: https://github.com/JuliaLang/julia/issues/20889
+[#20952]: https://github.com/JuliaLang/julia/issues/20952
 [#21183]: https://github.com/JuliaLang/julia/issues/21183
 [#21359]: https://github.com/JuliaLang/julia/issues/21359
 [#21692]: https://github.com/JuliaLang/julia/issues/21692
 [#21697]: https://github.com/JuliaLang/julia/issues/21697
+[#21746]: https://github.com/JuliaLang/julia/issues/21746
```
Comment (tkelman) [inline NEWS.md]: I think this and the rest of the new news entries are all 0.7, little cleaner to split these into a different commit

Comment (ararslan) [inline NEWS.md]: Oh, okay, will do. Those were all added by NEWS-update.

Comment (ararslan) [inline NEWS.md]: Done

**Review by Sacha0 — COMMENTED** [review]
Awesome! Thanks @ararslan! :)
